recipes/fast-dds/all/conanfile.py

```diff
@@ -91,6 +90,9 @@ def validate(self):
             # linking dynamic '*.dll' and static MT runtime
             raise ConanInvalidConfiguration("Mixing a dll {} library with a static runtime is a bad idea".format(self.name))
 
+    def build_requirements(self):
+        self.tool_requires("cmake/[>=3.16.3 <4]")
```

Comment:
```suggestion
        if Version(self.version) >= "2.7.0":
            self.tool_requires("cmake/[>=3.16.3 <4]")
```

https://github.com/eProsima/Fast-DDS/commit/acecac66b486c5948f237e332987bda8183ff4d4